### PR TITLE
Refresh extensions page

### PR DESCRIPTION
**Pull request by nickzhums**

### CODE REVIEW

**Comment by nickzhums** [comment]
@Eskibear @jdneo @testforstephen @CsCherrYY please help to review this page

**Comment by nickzhums** [comment]
I basically removed some out-of-date recommendations and re-structure extensions into more categorical sections instead of just a list

File: docs/java/extensions.md [hunk 1]
```diff
@@ -33,49 +28,63 @@ To help set up Java on VS Code, there is the Microsoft [Extension Pack for Java]
...
+<a class="tutorial-install-extension-btn" href="vscode:extension/Pivotal.vscode-boot-dev-pack">Install the Extension Pack for Java</a>
 
-Thanks to the great Java community around VS Code, there are many great community-created extensions as well. You can search for more Java extensions easily within the VS Code:
+More Spring Boot related information can be found at [Spring Boot](/docs/java/java-spring-boot) page.
 
-1. Go to the **Extensions** view (`kb(workbench.view.extensions)`).
-2. Filter the extensions list by typing 'java'.
+## Gradle for Java
+
+Visual Studio Code offers support Gradle support via the [Gradle for Java extension](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-gradle). To learn more how to use this extension, please visit the [Gradle section on Build Tools](/docs/java/java-build#_gradle) page
```
Comment (CsCherrYY) [inline docs/java/extensions.md:51]: how about [Gradle section](/docs/java/java-build#_gradle) on Build Tools page

File: docs/java/extensions.md [hunk 2]
```diff
@@ -9,20 +9,15 @@ MetaDescription: Popular Java extensions for Visual Studio Code
...
-1. Go to the **Extensions** view (`kb(workbench.view.extensions)`).
-2. Filter the extensions list by typing "java".
-
-![Java Extensions](images/extensions/extensions.png)
+## Fundamental Java Development
 
-## Recommended extensions
-
-To help set up Java on VS Code, there is the Microsoft [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), which contains these popular extensions:
+If you are looking for core Java development experience on Visual Studio Code (including Java code auto-completion, running / debugging / testing Java application, Java project management, etc), we recommend the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack). This extension pack contains a list of popular Java extensions for fundamental Java development:
```
Comment (jdneo) [inline docs/java/extensions.md:20]: Maybe add a `Install Extension Pack for Java` button like what we have in other docs?

Comment (nickzhums) [reply docs/java/extensions.md:20]: it's on line 29

File: docs/java/extensions.md (hunk 1 above)
Comment (nickzhums) [reply docs/java/extensions.md]: fixed

**Comment by Eskibear** [comment]
@CsCherrYY @jdneo please help to review the latest updates

**Comment by CsCherrYY** [comment]
LGTM👍

**Review by jdneo — APPROVED** [review]
LGTM

**Pull request merged**